Sources/TuistCore/Graph/GraphTraverser.swift

```diff
@@ -378,8 +378,12 @@ public class GraphTraverser: GraphTraversing {
         /// Other targets' frameworks.
         var otherTargetFrameworks = filterDependencies(
             from: .target(name: name, path: path),
-            test: isDependencyDynamicTarget,
-            skip: canDependencyEmbedBinaries
+            test: isEmbeddableDependencyTarget,
+            skip: { dependency in
+                testTarget(dependency: dependency) { target in
+                    canEmbedFrameworks(target: target) || target.product == .macro
+                }
+            }
```

Comment:
```suggestion
     or(canDependencyEmbedBinaries, isDependencyPrecompiledMacro)
```

The current condition is same as: https://github.com/tuist/tuist/pull/7006/files#diff-e92bb6ba816dd394aacf7d74e51dc2269fc21f4b2f1f27031ffede91837cb0e5R357

Reply:
Actually, it's not the same. What `isDependencyPrecompiledMacro` basically does is to check if `dependency == .macro`. Here we need to check if `dependency` is `target` and its `product` is `macro`

Reply:
Got it, thanks for the explanation.